Dialog/CreditEditor_TableView.cpp

```diff
@@ -0,0 +1,237 @@
+﻿#include "Dialog/CreditEditor_TableView.h"
+#include "Dialog/CreditsEditDialog.h"
+#include <QTableView>
+#include <QStyledItemDelegate>
+#include <QMenu>
+#include <iostream>
+#include <QPainter>
+#include <QKeyEvent>
+
+
+CreditEditor_TableView::CreditEditor_TableView(QWidget *parent) :
+    QTableView(parent)
+{
+    menu=new QMenu("Menu", this);
+    action_one_tile = new QAction("One Tile",menu);
+    action_upper_tile = new QAction("Upper Tile",menu);
+    action_lower_tile = new QAction("Lower Tile",menu);
+    action_delete = new QAction("Delete",menu);
+
+    menu->addAction(action_one_tile);
+    menu->addAction(action_upper_tile);
+    menu->addAction(action_lower_tile);
+    menu->addAction(action_delete);
+
+    this->setContextMenuPolicy(Qt::CustomContextMenu);
+    connect(this, SIGNAL(customContextMenuRequested(const QPoint &)), this, SLOT(showContextMenu(const QPoint &)));
+    connect(menu, SIGNAL(triggered(QAction *)), this, SLOT(doAction(QAction *)));
+}
+
+
+
+/// <summary>
+/// Called each time a cell is changed
+/// </summary>
+/// <param name="topLeft">
+/// The top left QModelIndex
+/// <param name="bottomRight">
+/// Unused
+/// <param name="roles">
+/// Unused
+/// </param>
+void CreditEditor_TableView::dataChanged(const QModelIndex &topLeft, const QModelIndex &bottomRight, const QVector<int> &roles) {
+
+    QString first_letter=topLeft.data(Qt::DisplayRole).value<QString>();
+    QString role=topLeft.data(Qt::BackgroundRole).value<QColor>().name();
+
+    std::string string = first_letter.toUpper().toUtf8().constData();
+
+    if (string.size() > 0) {
+        std::string role_string = role.toUtf8().constData();
+        std::string first_letter_string(1,string.at(0));
+
+        //If the tile is not valid
+        if (CreditsEditDialog::CreditTileReverseMap[std::make_pair(first_letter_string,role_string)] == 0) {
+
+            //If the tile is still not valid with the default background we delete it
+            if (CreditsEditDialog::CreditTileReverseMap[std::make_pair(first_letter_string,"#8080ff")] == 0) {
```

Comment:
Let's `#define` the color strings at the top of the file instead of hardcoding them